crates/gen/src/types/required_interface.rs

```diff
@@ -117,6 +117,10 @@ impl RequiredInterface {
                 kind = InterfaceKind::NonDefault;
             }
 
+            if interfaces.iter().any(|i| i.name == append_name) {
```

Comment:
Previously, duplicates were creeping in and causing invalid Rust to be generated with duplicate trait implementations for some structs (in some rarely used Xaml namespaces).

Reply:
I wonder if storing the interfaces in a HashMap would be better. It uses a tiny bit more memory, but the lookup for duplicates will be faster.